create_infotexts.py

```diff
@@ -233,39 +232,44 @@ def create_commons_filename(self, item):
         :param item: dictionary conatining metadata for one image
         :return: None (populates self.filename)
         """
-        fname_str = ""
-        fname_str += re.sub(" Svenska Cypernexpeditionen\.?", "", item["Beskrivning"])
-        if not fname_str.endswith(" "):
-            fname_str += " "
+        fname_desc = ""
+        fname_desc += re.sub(" Svenska Cypernexpeditionen\.?", "", item["Beskrivning"])
+        if not fname_desc.endswith("."):
+            fname_desc += "."
 
         # Enrich with keywords
+        keywords_to_append = []
         if self.data["keyword_list"]:
             for keyword in self.data["keyword_list"]:
                 if keyword not in item["Beskrivning"]:
-                    fname_str += keyword + " "
+                    keywords_to_append.append(keyword)
+
+        for kw in keywords_to_append:
+            fname_desc += kw + ", "
+        fname_desc.rstrip(", ")
 
         # Enrich with regional data
         if not item["Ort, foto"] in item["Beskrivning"]:
-            fname_str += item["Ort, foto"] + " "
+            fname_desc += item["Ort, foto"] + ", "
 
         if not item["Region, foto"] in item["Beskrivning"]:
-            fname_str += item["Region, foto"] + " "
+            fname_desc += item["Region, foto"] + ", "
 
         if not item["Land, foto"] in item["Beskrivning"]:
-            fname_str += item["Land, foto"] + " "
+            fname_desc += item["Land, foto"] + ", "
 
         # Ensure first descriptive part is not empty
-        if fname_str == " ":
-            fname_str = fname_str.replace(" ", "")
-            fname_str += "Svenska Cypernexpeditionen 1927-1931"
+        if fname_desc == " ":
+            fname_desc = fname_desc.replace(" ", "")
+            fname_desc += "Svenska Cypernexpeditionen 1927-1931"
 
         # Batch information
-        fname_str += "-_SMVK - MM - Cypern_ - _"
+        institution = "SMVK"
 
         # Ensure unique by adding <Fotonummer>
-        fname_str += item["Fotonummer"]
+        idno = item["Fotonummer"]
 
-        self.filename = fname_str
+        self.filename = helpers.format_filename(fname_desc, institution, idno)
```

Comment:
I would simply do it all here:

`helpers.format_filename(fname_desc, "SMVK", item["Fotonummer"])`